src/AutoRest.CSharp/Common/Generation/Writers/ClientOptionsWriter.cs

```diff
@@ -50,6 +51,21 @@ public static void WriteClientOptions(CodeWriter writer, ClientOptionsTypeProvid
                             }
                         }
                     }
+
+                    if (clientOptions.AdditionalProperties != null && !Configuration.KeepOptionalClientParametersInConstructor)
+                    {
+                        if (clientOptions.ApiVersions is not null)
+                        {
+                            writer.Line();
+                        }
+
+                        foreach (var property in clientOptions.AdditionalProperties)
```

Comment:
this is kind of misleading because the type of this `property` variable is actually `Parameter`.
Could we normalize it to be `var parameter in clientOptions.AdditionalParameters`?